CondTools/SiPhase2Tracker/plugins/DTCCablingMapProducer.cc

```diff
@@ -0,0 +1,299 @@
+{
+	using namespace std;
+	
+	for (std::string const& csvFileName : modulesToDTCCablingCSVFileNames)
+	{
+		edm::FileInPath csvFilePath(csvFileName);
+		
+		ifstream csvFile;
+		csvFile.open(csvFilePath.fullPath().c_str());
+		
+		if (csvFile.is_open())
+		{
+			string csvLine;
+			
+			unsigned lineNumber = 0;
+			
+			while (std::getline(csvFile, csvLine))
+			{
+				if (verbosity_ >= 1)
+				{
+					edm::LogInfo("CSVParser") << "Reading CSV file line: " << ++lineNumber << ": \"" << csvLine << "\"" << endl;
+				}
+				
+				istringstream csvStream(csvLine);
+				vector<string> csvColumn;
+				string csvElement;
+				
+				while (std::getline(csvStream, csvElement, ','))
+				{
+					trim(csvElement);
+					csvColumn.push_back(csvElement);
+				}
+				
+				if (verbosity_ >= 2)
+				{
+					ostringstream splitted_line_info;
+					
+					splitted_line_info << "-- split line is: [";
+					
+					for (string const& s : csvColumn)
+						splitted_line_info << "\"" << s << "\", ";
+					
+					splitted_line_info << "]" << endl;
+					
+					edm::LogInfo("CSVParser") << splitted_line_info.str();
+				}
+				
+				if (csvColumn.size() == csvFormat_ncolumns_)
+				{
+					// Skip the legend lines
+					if (0 == csvColumn[0].compare(std::string("Module DetId/U")))
+					{
+						if (verbosity_ >= 1)
+						{
+							edm::LogInfo("CSVParser") << "-- skipping legend line" << endl;
+						}
+						continue;
+					}
+					
+					uint32_t detIdRaw;
+					
+					try
+					{
+						detIdRaw = std::stoi( csvColumn[csvFormat_idetid_] );
+					}
+					catch (std::exception e)
+					{
+						if (verbosity_ >= 0)
+						{
+							edm::LogError("CSVParser") << "-- malformed DetId string in CSV file: \"" << csvLine << "\"" << endl;
+						}
+						throw e;
+					}
+					
+					unsigned const dtc_id = strtoul(csvColumn[csvFormat_idtcid_].c_str(), nullptr, 10);
+					unsigned gbt_id ;
+					unsigned elink_id;
+					
+					if (generate_fake_valid_gbtlink_and_elinkid_)
+					{
+						for (gbt_id = 0; gbt_id < 72; ++gbt_id)
+						{
+							for  (elink_id = 0; elink_id < 7; ++elink_id)
+							{
+								if ( !(pCablingMap_->knowsDTCELinkId( DTCELinkId(dtc_id, gbt_id, elink_id) )) )
+									goto gbtlink_and_elinkid_generator_end;
+							}
+						}
+						gbtlink_and_elinkid_generator_end:
+						((void)0); // This is a NOP, it's here just to have a valid (although dummy) instruction after the goto tag
+					}
+					else
+					{
+						gbt_id   = strtoul(csvColumn[csvFormat_igbtlinkid_].c_str(), nullptr, 10);
+						elink_id = strtoul(csvColumn[csvFormat_ielinkid_  ].c_str(), nullptr, 10);
+					}
+					
+					
+					DTCELinkId dtcELinkId( dtc_id, gbt_id, elink_id );
+					
+					if (verbosity_ >= 3)
+					{
+						edm::LogInfo("CSVParser") << "-- DetId = " << detIdRaw << " (dtc_id, gbt_id, elink_id) = (" << dtc_id << "," << gbt_id << "," << elink_id << ")" << endl;
+					}
+					
+					{
```

Comment:
This pair of curly braces seems to be redundant.

Reply:
Fixed. It was the second brace and the following one pairing with it that were redundant.